system-apps/desktop-documentation-viewer/webClient/src/App.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as React from 'react';
+import { MemoryRouter, Route } from 'react-router';
+import Main from './Main';
+import { withTranslation } from 'react-i18next';
+
+class App extends React.Component<any, any> {
+  private log: ZLUX.ComponentLogger;
+  t: any;
+  constructor(props){
+    super(props);
+    this.log = this.props.resources.logger;
+    let metadata = this.props.resources.launchMetadata;
+    if (metadata != null && metadata.data != null && metadata.data.type != null) {
+      this.handleLaunchOrMessageObject(metadata.data);
+    } else {
+      this.state = this.getDefaultState();
+    }
+
+  };
+
+  private getDefaultState() {
+    return {
+      actionType: "Launch",
+      appTarget: "PluginCreate",
+      docsList: "",
+      pluginIdentifier: "",
+      allPlugins: ZoweZLUX.pluginManager.loadPlugins("application", false).__zone_symbol__value,
+      destination: ZoweZLUX.uriBroker.pluginRESTUri(this.props.resources.pluginDefinition.getBasePlugin(), 'plugindetail', "")
+    };
+  }
+
+  updateOrInitState(stateUpdate: any): void {
+    if (!this.state) {
+      this.state = Object.assign(this.getDefaultState(), stateUpdate);
+    }
+    else {
+      this.setState(stateUpdate);
+    }
+  }
+
+  handleLaunchOrMessageObject(data: any) {
```

Comment:
this is for app2app communication to have 1 app to something another app asks for. you could use it to recieve a request to open a specific document, for example. But, it might be better to just remove it for now.